libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetryTest.kt

```diff
@@ -1,131 +1,613 @@
 package com.mapbox.navigation.core.telemetry
 
+import android.app.ActivityManager
 import android.app.AlarmManager
 import android.content.Context
 import android.content.SharedPreferences
+import android.location.Location
+import android.media.AudioManager
+import android.telephony.TelephonyManager
+import com.mapbox.android.telemetry.AppUserTurnstile
 import com.mapbox.android.telemetry.MapboxTelemetryConstants
 import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.api.directions.v5.models.StepManeuver
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.metrics.MetricEvent
 import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.base.trip.model.RouteLegProgress
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.base.trip.model.RouteProgressState.LOCATION_TRACKING
+import com.mapbox.navigation.base.trip.model.RouteProgressState.ROUTE_COMPLETE
+import com.mapbox.navigation.base.trip.model.RouteStepProgress
 import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.NavigationSession
+import com.mapbox.navigation.core.NavigationSession.State.ACTIVE_GUIDANCE
+import com.mapbox.navigation.core.NavigationSession.State.FREE_DRIVE
+import com.mapbox.navigation.core.NavigationSession.State.IDLE
+import com.mapbox.navigation.core.telemetry.events.NavigationArriveEvent
+import com.mapbox.navigation.core.telemetry.events.NavigationCancelEvent
+import com.mapbox.navigation.core.telemetry.events.NavigationDepartEvent
+import com.mapbox.navigation.core.telemetry.events.NavigationEvent
+import com.mapbox.navigation.core.telemetry.events.NavigationFeedbackEvent
+import com.mapbox.navigation.core.telemetry.events.NavigationRerouteEvent
 import com.mapbox.navigation.metrics.MapboxMetricsReporter
-import com.mapbox.navigation.testing.MainCoroutineRule
-import com.mapbox.navigation.utils.internal.ThreadController
+import com.mapbox.navigation.metrics.internal.event.NavigationAppUserTurnstileEvent
+import io.mockk.Runs
+import io.mockk.coEvery
 import io.mockk.every
+import io.mockk.just
 import io.mockk.mockk
 import io.mockk.mockkObject
 import io.mockk.unmockkObject
 import io.mockk.verify
-import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.InternalCoroutinesApi
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertNotSame
+import junit.framework.TestCase.assertTrue
 import org.junit.After
-import org.junit.Before
-import org.junit.Rule
 import org.junit.Test
 
-@InternalCoroutinesApi
-@ExperimentalCoroutinesApi
 class MapboxNavigationTelemetryTest {
```

Comment:
Same as above re: we normally organize the code in this particular order :point_right: `public`, `protected`, `private`, etc.

Reply:
done